src/cli/config/config.go

```diff
@@ -209,14 +220,24 @@ func writeRepoConfigWithViper(vpr *viper.Viper, s3Config storage.S3Config, m365C
 	return nil
 }
 
+// GetStorageAndAccount creates a storage and account instance by mediating all the possible
+// data sources (config file, env vars, flag overrides) and the config file.
+func GetStorageRepoAndAccount(ctx context.Context) (storage.Storage, account.Account, string, error) {
```

Comment:
Instead of a new func, can we introduce a struct like this?
```go
type configRepoDetails struct {
    Storage storage.Storage
    Account account.Account
    RepoID string
}
```
Then return the struct in `GetStorageAndAccount` (maybe rename it to GetConfigRepoDetails).  That way we maintain flexibility and extensibility without additional funcs or 4+ value returns.

Note that I think we should _return_ this struct when reading the config, but not _require_ it when writing the config with `WriteRepoConfig`